.github/workflows/ci.yml

```
@@ -2,24 +2,41 @@ name: Java CI with Maven
         uses: actions/checkout@v4
+
       - name: Set up JDK
         uses: actions/setup-java@v4
         with:
           distribution: 'temurin'
           java-version: '23'
           cache: maven
+
+      - name: Check for changes in Java source or test files
+        id: check_java_changes
+        run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2086:info:2:27: Double quote to prevent globbing and word splitting

(shellcheck)